timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -310,23 +312,38 @@ def _predict(
             )
             self.model_pipeline.model.eval()
             with torch.inference_mode():
-                prediction_samples = [
-                    self.model_pipeline.predict(
-                        batch,
-                        prediction_length=self.prediction_length,
-                        num_samples=self.num_samples,
-                        limit_prediction_length=False,
-                    )
-                    .detach()
-                    .cpu()
-                    .numpy()
-                    for batch in inference_data_loader
-                ]
-
-        samples = np.concatenate(prediction_samples, axis=0).swapaxes(1, 2).reshape(-1, self.num_samples)
-
-        mean = samples.mean(axis=-1, keepdims=True)
-        quantiles = np.quantile(samples, self.quantile_levels, axis=-1).T
+                if self.model_pipeline.forecast_type == ForecastType.SAMPLES:
+                    prediction_samples = [
```

Comment:
Maybe we can put the inference call/quantile computation in a separate function for clarity?